Tasks/Common/docker-common/registryauthenticationprovider/acrauthenticationtokenprovider.ts

```diff
@@ -0,0 +1,37 @@
+"use strict";
+
+import * as tl from "vsts-task-lib/task";
+import RegistryAuthenticationToken from "./registryauthenticationtoken"
+import AuthenticationTokenProvider from "./authenticationtokenprovider"
+
+export default class ACRAuthenticationTokenProvider extends AuthenticationTokenProvider{
+
+    // URL to registry like jitekuma-microsoft.azurecr.io
+    private registryURL: string;
+
+    // name of the azure subscription endpoint like RMDev
+    private endpointName: string;
+
+    // ACR fragment like /subscriptions/c00d16c7-6c1f-4c03-9be1-6934a4c49682/resourcegroups/jitekuma-RG/providers/Microsoft.ContainerRegistry/registries/jitekuma
+    private acrFragmentUrl: string;
```

Comment:
We need  registryURL, endpointName but acrFragmentUrl is extra but i kept it as this is unique url for the registry in ACR and i want to keep it, in case we require in future to make rest call. Currently we are not using acrFragmentUrl.